Classes/TCA/FormHelper.php

```diff
@@ -93,62 +96,56 @@ protected function findUserByGroupRoleAndLocation($role, $location)
 
         $groupsArray = [];
 
-        $where = 'be_groups.tx_ats_location = "'.$location.'"';
+        $queryBuilder = $this->getQueryBuilder('be_groups');
+        $queryBuilder->select('uid')
+            ->from('be_groups')
+            ->where($queryBuilder->expr()->eq('tx_ats_location', $queryBuilder->createNamedParameter($location)));
 
         // Include extbase acl roles if available
         if (ExtensionManagementUtility::isLoaded('extbase_acl') && !empty($role)) {
-            $where .= ' AND be_groups.tx_extbaseacl_role = "' . $role . '"';
+            $queryBuilder->andWhere(
+                $queryBuilder->expr()->eq('tx_extbaseacl_role', $queryBuilder->createNamedParameter($role))
+            );
         }
 
-        $where .= $this->getDeleteClause("be_groups") . $this->getBackendEnableFields("be_groups");
-
+        $res = $queryBuilder->execute();
 
-        $res = $GLOBALS['TYPO3_DB']->exec_SELECTquery("uid", "be_groups", $where);
-        while ($row = $GLOBALS['TYPO3_DB']->sql_fetch_assoc($res)) {
+        while ($row = $res->fetch()) {
             $groupsArray[] = $row['uid'];
         }
 
-        $where = ' ( 1=0 ';
+        $queryBuilder = $this->getQueryBuilder('be_users');
+
+        $queryBuilder->select("*")->from("be_users")->orderBy('realName');
+
+        $orWhere = [];
         foreach ($groupsArray as $val) {
-            $where .= " OR FIND_IN_SET($val, `usergroup`)";
+            $orWhere[] = $queryBuilder->expr()->inSet('usergroup', $queryBuilder->createNamedParameter($val));
         }
-        $where .= ')';
 
-        $where .= $this->getDeleteClause("be_users") . $this->getBackendEnableFields("be_users");
+        if (!empty($orWhere)) {
+            $queryBuilder->orWhere(...$orWhere);
+        }
 
-        $orderBy = 'realName';
-        $res = $GLOBALS['TYPO3_DB']->exec_SELECTquery("*", "be_users", $where, '', $orderBy);
+        $res = $queryBuilder->execute();
 
         $items = [];
 
-        while ($row = $GLOBALS['TYPO3_DB']->sql_fetch_assoc($res)) {
+        while ($row = $res->fetch()) {
             $items[] = array($row['realName'].' ('.$row['username'].')', $row['uid']);
         }
-
         return $items;
     }
 
     /**
-     * Helper function to mock static BackendUtility function
-     * @codeCoverageIgnore
-     *
-     * @param  string $table
-     * @return string
-     */
-    public function getDeleteClause($table)
-    {
-        return BackendUtility::deleteClause($table);
-    }
-
-    /**
-     * Helper function to mock static BackendUtility function
+     * Helper function to mock QueryBuilder
      * @codeCoverageIgnore
      *
      * @param  string $table
-     * @return string
+     * @return object
      */
-    public function getBackendEnableFields($table)
+    public function getQueryBuilder($table)
```

Comment:
Do we really need this helper function? You could instance ConnectionPool once and use it directly.